core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -661,4 +669,51 @@ public boolean equals(Object obj) {
       }
    }
 
+   public void testLockedStream() throws Throwable {
+      for (int i = 0; i < 10; i++) {
+         cache.put(i, "value" + i);
+      }
+
+      CyclicBarrier barrier = new CyclicBarrier(2);
+
+      int key = 4;
+
+      LockedStream<Object, Object> stream = cache.getAdvancedCache().lockedStream();
+      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
+         if (e.getKey().equals(key)) {
+            try {
+               barrier.await(10, TimeUnit.SECONDS);
+               assertEquals("value" + key, c.put(e.getKey(), String.valueOf(e.getValue() + "-other")));
+
+               barrier.await(10, TimeUnit.SECONDS);
+            } catch (InterruptedException | BrokenBarrierException | TimeoutException e1) {
+               throw new RuntimeException(e1);
+            }
+         }
+      }));
+
+      barrier.await(10, TimeUnit.SECONDS);
+
+      Future<Object> putFuture = fork(() -> cache.put(key, "value" + key + "-new"));
+      try {
+         putFuture.get(50, TimeUnit.MILLISECONDS);
```

Comment:
Sorry there are so many `expectException` methods on the class, I didn't realize there is a `ExceptionRunnable` override one. I was looking at the first ones, which would be considerably more code. I personally feel the try/catch is more readable and direct than the `ExceptionRunnable`, but I can change it to that.